server/lib/form-prepare.js

```diff
@@ -0,0 +1,46 @@
+import ClientError from './client-error.js';
+
+/**
+ * Creates an array of all the items values from the request object.
+ * @param {object} req, the request.
+ * @param {number} numberOfItems, total number of items to be extracted.
+ * @param {number} recordId, the foreign key to attach all the items to.
+ * @returns an array of items taken from the request body.
+ */
+function createItemsList(req, numberOfItems, recordId) {
+  const items = [];
+  for (let i = 0; i < numberOfItems; i++) {
+    items.push(recordId, req.body[`item${i}`], req.body[`itemCat${i}`], req.body[`itemAmt${i}`]);
+  }
+  if (items.includes(undefined) || items.includes(null)) throw new ClientError(400, 'Incomplete form.');
+  return items;
+}
+
+/**
+ * Creates the SQl for inserting a dynamic amount of rows into the items table.
```

Comment:
This could use more explanation, such as providing an example. I read this code several times before I figured out what it was doing, and why.